dataclasses_avroschema/utils.py

```diff
@@ -80,6 +80,9 @@ def standardize_custom_type(value: typing.Any) -> typing.Any:
         return tuple(standardize_custom_type(v) for v in value)
     elif issubclass(type(value), enum.Enum):
         return value.value
+    elif callable(getattr(value, "asdict", None)):
```

Comment:
is it possible to use the function [is_pydantic_model](https://github.com/marcosschroh/dataclasses-avroschema/blob/master/dataclasses_avroschema/utils.py#L21) instead?

Reply:
It seems it does not work - `is_pydantic_model` works only for classes definition and raises 
```
TypeError: issubclass() arg 1 must be a class
```
error. Either we modify this function or leave the check as is :)

Reply:
`is_pydantic_model` checks for `pydantic.BaseModel` which has a `dict()` method but not `asdict()`
However, with that check this part can be simplified to
```suggestion
    elif is_pydantic_model(value):
        return standardize_custom_type(value.dict())
```

        ```

Reply:
yeah, but AvroBaseModel value also is True on `is_pydantic_model` and we would call then `dict()` which does not handle all serialization recusion, right? This is why we have separate `asdict()`.

We could check for both methods and run `asdict()` with bigger priority.

Reply:
Exactly. We should always use the method `asdict()` which will make sure that all the data is standardize properly. If we use the `dict` method (pydantic) the data won't be properly standarized and then the `avro serializarion` will fail.

Reply:
> This is why we have separate asdict().

If you notice, asdict jsut calls dict(self) and then calls this function.
I think calling dict() here and recursive pass the value to this function reproduces the same behavior.
But don't trust me, trust the unit tests, that's why we have them :) If the unit test demonstrate the intended behavior then it's problem solved.